pkg/gardener/shoot/extender/extensions/network_filter.go

```diff
@@ -0,0 +1,78 @@
+package extensions
+
+import (
+	"encoding/json"
+	gardener "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryRuntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const NetworkFilterType = "shoot-networking-filter"
+
+func NewNetworkFilterExtension(filter imv1.Filter) (*gardener.Extension, error) {
+	disabled := isNetworkingFilterDisabled(filter)
+
+	networkingFilterExtension := gardener.Extension{
+		Type:     NetworkFilterType,
+		Disabled: &disabled,
+	}
+
+	if disabled {
+		return &networkingFilterExtension, nil
+	}
+
+	if isIngressBlackholingEnabled(filter) {
+		ingressFilterConfig := filter.Ingress
+
+		filterProviderConfig := Configuration{
+			TypeMeta: metav1.TypeMeta{},
+			EgressFilter: &EgressFilter{
+				BlackholingEnabled: ingressFilterConfig.Enabled,
```

Comment:
Can be even more simplified with 
```suggestion
				BlackholingEnabled: true,
```